inc/saimacsec.h

```diff
@@ -0,0 +1,1478 @@
+
+    /**
+     * @brief End of Rule Match Fields
+     */
+    SAI_MACSEC_RULE_ATTR_FIELD_END = SAI_MACSEC_RULE_ATTR_FIELD_MACSEC_SCI_ID,
+
+    /**
+     * @brief Enable/disable control of the MACsec rule.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_MACSEC_RULE_ATTR_RULE_ENABLE,
+
+    /**
+     * @brief MACSEC flow object id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_MACSEC_FLOW
+     */
+    SAI_MACSEC_RULE_SECURE_CHANNEL_ATTR_FLOW_ID,
+
+    /**
+     * @brief End of MACSEC Rule attributes
+     */
+    SAI_MACSEC_RULE_ATTR_END,
+} sai_macsec_rule_attr_t;
+
+/**
+ * @brief MACSEC rule counter ID in sai_get_macsec_rule_stats_ext() call
+ */
+typedef enum _sai_macsec_rule_stat_t
+{
+   /**
+    *@brief Count of packets that hit this rule
+    */
+    SAI_MACSEC_RULE_STAT,
+
+} sai_macsec_rule_stat_t;
+
+/**
+ * @brief Attribute Id for sai_macsec_secure_channel
+ *
+ * @flags Contains flags
+ */
+typedef enum _sai_macsec_secure_channel_attr_t
+{
+    /**
+     * @brief Start of MACSEC Secure Channel attributes
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_START,
+
+    /**
+     * @brief MACSEC direction
+     *
+     * @type sai_macsec_direction_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_DIRECTION = SAI_MACSEC_SECURE_CHANNEL_ATTR_START,
+
+    /**
+     * @brief MACSEC flow object id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_MACSEC_FLOW
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_FLOW_ID,
+
+    /**
+     * @brief SCI value for this Secure Channel, carried in MACSEC packet SecTAG.
+     *
+     * @type sai_macsec_sci_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_SCI,
+
+    /**
+     * @brief SSCI value for this Secure Channel
+     *
+     * @type sai_macsec_ssci_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_SSCI,
+
+    /**
+     * @brief XPN enable for this Secure Channel
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_XPN_ENABLE,
+
+    /**
+     * @brief Explicit SCI enable for this Secure Channel.
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @condition SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_DIRECTION == SAI_MACSEC_DIRECTION_EGRESS
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_EXPLICIT_SCI_ENABLE,
+
+    /**
+     * @brief SecTAG offset for this Secure Channel.
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @condition SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_DIRECTION == SAI_MACSEC_DIRECTION_EGRESS
+     */
+    SAI_MACSEC_SECURE_CHANNEL_ATTR_MACSEC_SECTAG_OFFSET,
+
+    /**
+     * @brief Active MACSEC secure_association
+     * @type sai_object_id_t
+     * @flags SET_ONLY
```

Comment:
there is no such flag, create_only you mean?, when using condition then mandatory_on_create must be used